### PR TITLE
implement AliasAction and add small tests

### PR DESCRIPTION
**Pull request by yongpingchen**
- Implement AliasAction, make it extends Parcelable.
- Add small tests

### CODE REVIEW

File: thingif/src/main/java/com/kii/thingif/command/AliasAction.java [hunk 1]
```diff
@@ -1,19 +1,101 @@
...
+        @Override
+        public AliasAction[] newArray(int size) {
+            return new AliasAction[size];
+        }
+    };
+
+    @Override
+    public boolean equals(Object o) {
+        if (o == null) return false;
+        if (!(o instanceof AliasAction)) return false;
```
Comment (shinsuke-nara) [inline thingif/src/main/java/com/kii/thingif/command/AliasAction.java]:
```java
 if (!(o instanceof AliasAction)) return false;
```

This allows sub class of `AliasAction`. Is it your intention? If No, following code may be better:

```java
if (o.getClass() = AliasAction.class) return false;
```

What do you think?

File: thingif/src/test/java/com/kii/thingif/command/AliasActionTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        for(int i=0; i<sameObjects.length; i++) {
+            Assert.assertEquals(
+                    "failed to test equals for ["+i+"]",
+                    aliasAction,
+                    sameObjects[i]);
+
+            Assert.assertEquals(
+                    "failed to test hashCode for ["+i+"]",
+                    aliasAction,
+                    sameObjects[i]);
```
Comment (shinsuke-nara) [inline thingif/src/test/java/com/kii/thingif/command/AliasActionTest.java]: You want to check `aliasAction.hashCode()` don't you?

File: thingif/src/test/java/com/kii/thingif/command/AliasActionTest.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        new AliasAction<>("alias", (Action)null);
+    }
+
+    @Test
+    public void equals_hashCodeTest() {
+        AirConditionerActions action = new AirConditionerActions(true, null);
+        AliasAction<AirConditionerActions> aliasAction =
+                new AliasAction<>("alias", action);
+
+        Object[] sameObjects = {
```
Comment (shinsuke-nara) [inline thingif/src/test/java/com/kii/thingif/command/AliasActionTest.java]: I do not understand why an array of Objects is required.

**Comment by shinsuke-nara** [comment]
@yongpingchen Could you check my comment?

File: thingif/src/main/java/com/kii/thingif/command/AliasAction.java (hunk 1 above)
Comment (yongpingchen) [inline thingif/src/main/java/com/kii/thingif/command/AliasAction.java]: > This allows sub class of AliasAction.

That is true. 
I follow the guide line about implementation of `equals` in `Effective JAVA`,  it says use `getClass` will broke `Lislov substitution principle`.

File: thingif/src/test/java/com/kii/thingif/command/AliasActionTest.java (hunk 3 above)
Comment (yongpingchen) [inline thingif/src/test/java/com/kii/thingif/command/AliasActionTest.java]: self and same object are in the array to test

File: thingif/src/test/java/com/kii/thingif/command/AliasActionTest.java (hunk 2 above)
Comment (yongpingchen) [inline thingif/src/test/java/com/kii/thingif/command/AliasActionTest.java]: My mistake.

**Comment by shinsuke-nara** [comment]
@yongpingchen LGTM

**Pull request merged**